horace_core/sqw/file_io/@horace_binfile_interface/private/convert_old_det_forms_.m

```diff
@@ -0,0 +1,44 @@
+function det = convert_old_det_forms_(detpar,n_instances)
+% Method used to convert detector formats written by any previous version
+% of Horace into Horace 4.01 data form.
+%
+% Input:
+% detpar      -- an old or new format detector information.
+%                Normally obtained from binary sqw file.
+% n_instances -- number of run, this
+%
+% Returns:
+% det         -- detector information packed in
+%                unique_object_container container and
+%                distributed into approriate number of input
+%                runs.
+
+if isstruct(detpar) % we do not need do deal with struct array
+    % here as this have never been used and stored in
+    detpar = IX_detector_array(detpar);
+end
+if isa(detpar,'IX_detector_array')
+    det = unique_objects_container('baseclass','IX_detector_array');
+    det = det.add(detpar);
+    det = det.replicate_runs(n_instances);
+elseif isa(detpar,'unique_objects_container') || isa(detpar,'unique_references_container')
+    if detpar.n_objects == 1
+        det = detpar.replicate_runs(n_instances);
+    elseif ~isempty(n_instances) && detpar.n_objects == n_instances
```

Comment:
does this deal with _instances==0 which is non empty

Reply:
added checks and tests for '0' to deal with it reasonably